lib/watchers/periodic-watcher.js

```diff
@@ -0,0 +1,299 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright (c) 2015, Joyent, Inc.
+ */
+
+/*
+ * This module exists to allow consumers to get events when VM changes occur on
+ * a CN. It works by running 'vmadm lookup -j -o ...' periodically and comparing
+ * the output for the set of fields defined. When a change is noticed, it calls
+ * the opts.updateVm function that is passed in.
+ *
+ * This function is described in more detail in lib/vm-watcher.js.
+ *
+ */
+
+var assert = require('assert-plus');
+var diff = require('deep-diff').diff;
+var vmadm = require('vmadm');
+
+
+/*
+ * To just detect whether there has been a change to a VM we don't need to load
+ * all the fields each time. Any fields we don't load save some overhead in both
+ * gathering and processing. Most updates will modify last_modified for the VM,
+ * so mostly it is the fields that don't which we need to load. These include:
```

Comment:
Do we mean that updates that modify `last_modified` will be handled by the `fs-watcher` watcher? If so it would be clearer to mention that in this comment.